plugin/goto.py

```diff
@@ -15,53 +15,74 @@
+        fallback: bool = False
+    ) -> bool:
+        return fallback or super().is_enabled(event, point)
 
     def run(
         self,
         _: sublime.Edit,
         event: Optional[dict] = None,
         point: Optional[int] = None,
-        side_by_side: bool = False
+        side_by_side: bool = False,
+        fallback: bool = False
     ) -> None:
         session = self.best_session(self.capability)
         position = get_position(self.view, event, point)
         if session and position is not None:
             params = text_document_position_params(self.view, position)
             request = Request(self.method, params, self.view, progress=True)
-            session.send_request(request, functools.partial(self._handle_response_async, session, side_by_side))
+            session.send_request(
+                request, functools.partial(self._handle_response_async, session, side_by_side, fallback)
+            )
+        else:
+            self._handle_no_results(fallback, side_by_side)
 
     def _handle_response_async(
         self,
         session: Session,
         side_by_side: bool,
+        fallback: bool,
         response: Union[None, Location, List[Location], List[LocationLink]]
     ) -> None:
         if isinstance(response, dict):
             self.view.run_command("add_jump_record", {"selection": [(r.a, r.b) for r in self.view.sel()]})
             open_location_async(session, response, side_by_side)
         elif isinstance(response, list):
             if len(response) == 0:
-                window = self.view.window()
-                if window:
-                    window.status_message("No results found")
+                self._handle_no_results(fallback, side_by_side)
             elif len(response) == 1:
                 self.view.run_command("add_jump_record", {"selection": [(r.a, r.b) for r in self.view.sel()]})
                 open_location_async(session, response[0], side_by_side)
             else:
                 self.view.run_command("add_jump_record", {"selection": [(r.a, r.b) for r in self.view.sel()]})
                 sublime.set_timeout(functools.partial(LocationPicker, self.view, session, response, side_by_side))
         else:
-            window = self.view.window()
-            if window:
-                window.status_message("No results found")
+            self._handle_no_results(fallback, side_by_side)
+
+    def _handle_no_results(self, fallback: bool = False, side_by_side: bool = False) -> None:
+        window = self.view.window()
+
+        if not window:
+            return
+
+        if fallback and self.fallback_command:
+            window.run_command(self.fallback_command, {"side_by_side": side_by_side})
+        else:
+            window.status_message("No results found")
 
 
 class LspSymbolDefinitionCommand(LspGotoCommand):
     method = "textDocument/definition"
     capability = method_to_capability(method)[0]
+    fallback_command = "goto_definition"
```

Comment:
I haven’t followed the conversation in detail, but why shouldn’t the other goto variants have this fallback?

Reply:
I added it only for the `LspSymbolDefinitionCommand` because other commands don't have a Sublime equivalent.
In the current implementation, all the LSP specific commands(except `LspSymbolDefinitionCommand`)  will return `No results found` which makes sense to me